src/main/java/com/minecolonies/coremod/configuration/ConfigurationHandler.java

```diff
@@ -123,7 +136,22 @@ public static synchronized void init(final File file)
         }
         finally
         {
-            config.save();
+            if(config.hasChanged())
+                config.save();
+        }
+    }
+    
+    /**
+     * This event will be called when the config gets changed through
+     * the in-game GUI.
+     */
+    @SubscribeEvent
+    public void onConfigChanged(OnConfigChangedEvent eventArgs) 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document the parameter(s): eventArgs [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)